applications/OkAlice-Liquid-Staking.md

```diff
@@ -0,0 +1,153 @@
+# Liquid Staking - okAlice
+
+- **Team Name:** okAlice LS
+- **Payment Address:** 1xgDfXcNuB94dDcKmEG8rE9x9JVoqozCBnnitkN9nAe3Nyx
+
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+okAlice is on a mission to be at the forefront of building the Ink! ecosystem.
+The goal of this project is to create an Ink!-based liquid staking solution for Relay chain native tokens with the following functionality:
+
+- Enable users to participate in securing the network by staking their tokens.
+- Provide rewards for the staked tokens.
+- Offer a derivative token in exchange for the staked token and thus allowing users to utilize the derivative tokens on other dapps or chains, hence the name "liquid staking."
+
+### Ink! Ecosystem Impact
+
+Our Dapp is designed to bring substantial value to the Ink! ecosystem by implementing this DEFI solution on Astar. We expect that this implementation will attract a new influx of users, promoting growth and active participation within the Ink! community. One of our main goals is to remove the need for users to have technical knowledge about the Dapp's backend processes. Our Dapp will offer a straightforward liquid staking solution that can be completed with just a few clicks to stake the tokens.
+
+To facilitate knowledge sharing and inspire future projects, we are committed to making all our code open-source. Moreover, we intend to document our development process through articles and videos, ensuring transparency and easy access for everyone. Our goal is to create an inspiring point of reference for other projects, complementing the existing smart contracts within the Ink! ecosystem.
+
+### Project Details
+
+#### Architecture 
+
+![image](https://github.com/ok-Alice/Ecosystem-Grants/assets/18469570/249ae6ea-e83a-4765-9137-0fc91cc2b48e)
+
+Runtime Components:
+
+- **XCM CE:** the XCM chain-extension facilitates all contract interactions with the nomination pool on the relay chain. This chain-extension will be utilized during the development phase of the project. In the final release, we plan to leverage the native XCM support in Ink!.
+
+Ink! Contracts:
+
+- **issuer-staker:** Implements an asset with PSP22 interface for the derivative token. It handles staking operations on the nomination pool.
+- **Validator Selector:** Provides optimized management of the nomination pool based on the validator statistics published by the *Oracle*.
```

Comment:
We know that some off-chain scripts will be needed for house-hold tasks within the application. At this point we have not yet decided how or where to host them. We might reuse existing infrastructure for this purpose or look into other options such as Amazon Lambda.

Reply:
Happy to have a chat on this. We are working with a team developing Subquery / Subsquid oracle that can reports arbitrary indexed data to ink contracts.

Reply:
I will be happy to provide you a decentralized oracle built on phala network. 
I am already building on it and I did good progress.
I applied for a grant for this and I hope it will be accepted